src/vuln_analysis/tools/binary_tools.py

```
@@ -229,14 +261,19 @@ async def check_distro_patches(package_name: str, cve_id: str) -> str:
                 if not changelogs:
                     changelogs = list(changelog_dir.glob(f"*{package_name}*/changelog.Debian.gz"))
                 for changelog in changelogs[:3]:
+                    try:
+                        resolved_changelog = resolve_within_filesystem_root(changelog)
+                    except (RuntimeError, ValueError) as exc:
+                        evidence_parts.append(f"Refusing unsafe changelog path: {exc}")
+                        continue
                     try:
                         result = subprocess.run(
```

Ruff (0.16.1)

[warning] Async functions should not run processes with blocking methods

(ASYNC221)